### PR TITLE
Initial Map Support

### PR DESCRIPTION
**Pull request by wbaldoumas**
## Description

Begins addressing issue https://github.com/wayfair-incubator/AeroSharp/issues/9.

Begin implementation of initial map support, starting with just the following map functionality:

- `PutAsync`
- `GetByKeyAsync`
- `RemoveByKeyAsync`
- `DeleteAsync`

An `IMap` is provided for interacting with a single map. An `IMapOperator` is also provided for accessing various maps containing the same types.

The `MapBuilder` follows existing patterns for building an `IMap` or `IMap` operator.

[Aerospike scalar data types](https://docs.aerospike.com/server/guide/data-types/scalar-data-types) (64-bit integer and floating point numbers, booleans, and strings), as well as custom serialization of complex types are supported for map values.

Examples and unit tests have also been included for new functionality.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair-incubator/AeroSharp/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs
Comment (farmerau) [inline src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs:42]: So I pulled this into SharpLab, comparing `Any` vs `Count` vs pattern matching:

image

`Any` relies on Linq which is known to cause allocations, which we might want to avoid on a hot code path-- which this likely is. Comparing count against 0 has no allocations and communicates similarly the intent.

Pattern matching actually does the same thing, but also guarantees not being null, which you handled on line 35.

I imagine we're using Any in a number of places throughout this library-- so we could opt to continue for consistency. It might be another opportunity for a new issue to address this in the codebase as a whole.

File: src/AeroSharp/DataAccess/MapAccess/Parsers/MapParserWithSerializer.cs
Comment (farmerau) [inline src/AeroSharp/DataAccess/MapAccess/Parsers/MapParserWithSerializer.cs:70]: I realize this method doesn't know what the bin name is, but I think the error might be helpful if it did. This is similar to ExtractValue, I think. What do you think?

File: src/AeroSharp/DataAccess/MapConfiguration.cs
Comment (farmerau) [inline src/AeroSharp/DataAccess/MapConfiguration.cs:25]: The naming of `Partial` here is a little ambiguous. I see in the comment it's for writes/commits, but it's not immediately obvious in reading the property.

Could we do something like `AllowPartialWrites` or `AllowPartialCommits`? (Potentially also for `NoFail`?)

File: src/AeroSharp/DataAccess/Policies/MapConfigurationToMapPolicyMapper.cs
Comment (farmerau) [inline src/AeroSharp/DataAccess/Policies/MapConfigurationToMapPolicyMapper.cs:16]: Are these values mutually exclusive?

Comment (farmerau) [reply src/AeroSharp/DataAccess/Policies/MapConfigurationToMapPolicyMapper.cs:16]: If so, we may want to add a rule to the validator

File: src/AeroSharp/MapBuilder.cs
Comment (farmerau) [inline src/AeroSharp/MapBuilder.cs:39]: Nit: is the blank line here intentional?

**Review by farmerau — COMMENTED** [review]
Initial thoughts, haven't had a chance to look at the tests yet

File: examples/AeroSharp.Examples/Program.cs
Comment (awesor) [reply examples/AeroSharp.Examples/Program.cs:3]: It _is_ in our readme though for Integration Tests: https://github.com/wayfair-incubator/AeroSharp/blob/main/tests/AeroSharp.IntegrationTests/README.md

Perhaps we should add the same section to a new file in the Examples folder?

File: src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs
Comment (awesor) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: I guess it depends on how serialized this becomes, but the documentation was pretty clear I thought on what keys can be:

Map keys can be of type String, Bytes, Integer or Double.

So in this case, @farmerau, are you suggesting adding the ability to serialize a key?  I think a large part of the performance gain is just sticking to those keys and not adding more complexity at first here.

Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: My concern is we have no type constraint on `TKey`.

I’m not sure what `Value.Get` is going to return for a non immediately convertible type, but I’m hoping it’s not the HashCode.

So it’d be good to know ahead of time what the behavior might be (it might be covered in the tests I haven’t looked at yet) and more importantly that the keys are deterministic for arbitrary (value equivalent) objects.

File: src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs
Comment (awesor) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs:42]: I vote for starting this as the "new way" and if @farmerau doesn't mind opening an issue for the rest to replace any .Any's lingering around?

File: src/AeroSharp/DataAccess/MapConfiguration.cs
Comment (awesor) [reply src/AeroSharp/DataAccess/MapConfiguration.cs:25]: Yeah, to be fair, Aerospike themselves doesn't make this super clear either on their end: https://github.com/aerospike/aerospike-client-csharp/blob/80f15b7ffa44af30a8f4e1eeb0a28a1e7cada9a5/AerospikeClient/CDT/MapWriteFlags.cs

But I agree with @farmerau here - definitely would be good to make this a bit clearer for clients when setting it on their end.

File: src/AeroSharp/DataAccess/MapConfiguration.cs
Comment (awesor) [inline src/AeroSharp/DataAccess/MapConfiguration.cs:8]: nit: html

File: examples/AeroSharp.Examples/Maps/MapCustomClassExample.cs
Comment (wbaldoumas) [reply examples/AeroSharp.Examples/Maps/MapCustomClassExample.cs:10]: I'd love to get the repo swapped to file-scoped namespaces as a whole, since I think it really improves readability. I also think that file-scoped namespaces will be the default in the future. 

What are your thoughts on leaving this as-is and having a follow up task to swap to file-scoped namespaces across the repo?

File: examples/AeroSharp.Examples/Maps/MapCustomClassExample.cs
Comment (wbaldoumas) [reply examples/AeroSharp.Examples/Maps/MapCustomClassExample.cs:15]: Yeah, I can unseal the examples since it doesn't really provide any benefit here.

File: examples/AeroSharp.Examples/Program.cs
Comment (wbaldoumas) [reply examples/AeroSharp.Examples/Program.cs:3]: Yup! I didn't intend to remove this. Seems like the translation to using a top-level statement did it on it's own, so I'll add this back. 👍🏻

File: src/AeroSharp/DataAccess/Internal/RecordOperator.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/Internal/RecordOperator.cs:5]: We've disabled the requirement for system using directives to be placed first: https://github.com/wayfair-incubator/AeroSharp/blob/main/stylecop.json#L5

We've retained this rule to enforce using directives to be ordered alphabetically: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1210.md

I think it's up for debate in terms of what is the "right" way, but for now this is in line with the rest of the repo.

File: src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: I actually had support for opt-in key serialization to allow for arbitrarily complex keys, including associated tests, but ripped it out since I wasn't sure there would be a use case for it. Based on this convo it sounds like it might be nice to include.

I can add that back in, or do type checking of `TKey` within the builder to enforce simple keys. Thoughts?

File: src/AeroSharp/DataAccess/MapAccess/MapOperator.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/MapOperator.cs:56]: > Is this exception not being thrown by the record operator already? 

Interestingly enough, this isn't thrown for map operations. I confirmed by stepping through some of the integration test cases.

> I'm worried that by comparing against null here we're opening ourselves up to a situation where someone deliberately writes null as a value.

When someone writes a `null` value to a map, the map record itself will exist and `record` won't be `null` here. The value for the specific key that was written to in the map will be `null`. That said, I should probably add some test cases for this, so this is a good call out. Given that, do you have any other concerns here?

File: src/AeroSharp/DataAccess/MapAccess/Parsers/IMapBinParser.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/IMapBinParser.cs:17]: Wasn't sure what the best way to share this was, so here's quick screenshot.

image

What's stored on the returned map record seems to always be `KeyValuePair<object, object>` regardless of what types you've written to the map. I don't love it and didn't have an idea of what would be better here. What do you think?

File: src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs:42]: Nice find! I'm also a fan of pattern matching due to its expressiveness. I'll swap to that here.

+1 to @awesor's suggestion that we open up an issue for identifying and replacing other usages of `Any`.

File: examples/AeroSharp.Examples/Maps/MapCustomClassExample.cs
Comment (farmerau) [reply examples/AeroSharp.Examples/Maps/MapCustomClassExample.cs:10]: Sounds good to me.

File: src/AeroSharp/DataAccess/Internal/RecordOperator.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/Internal/RecordOperator.cs:5]: Sounds good.

File: src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: I think simple key enforcement is probably leading the user to the happiest path. I realize we can't do "OR" type constraints (bummer, but makes sense) so we'd need to handle it at runtime.

I think offering complex objects could be user friendly but might lead folks down a path of worse-than-necessary performance.

File: src/AeroSharp/DataAccess/MapAccess/MapOperator.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/MapOperator.cs:56]: Addressing the points in order:

1. Huh, that is interesting. Good catch and thanks for testing.
2. That makes sense. That addresses my set of concerns.

File: src/AeroSharp/DataAccess/MapAccess/Parsers/IMapBinParser.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/IMapBinParser.cs:17]: Huh, maybe something that would need addressed lower in the core library instead of at the Aerosharp level. Though, I'm not confident that any solution would be backwards compatible.

Sad times, but thanks for checking-- I figured you did it for good reason.

File: src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs:42]: Yep, I'll open up an issue for it.

Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs:42]: Scratch that. Since we're multi-targeting and supporting .NET 6.0, which only supports up to C# 10, we can't use that form of pattern matching since it's a C# 11 feature. We'd need to do something funky like this, which I don't love:

```csharp
#if NET6_0
        if (binValues.Count == 0)
        {
            throw new MapEntryNotFoundException($"Map entry not found in bin \"{bin}\".");
        }
#endif

#if NET7_0
        if (binValues is [])
        {
            throw new MapEntryNotFoundException($"Map entry not found in bin \"{bin}\".");
        }
#endif
```

That said, I'm going to use the count check.

Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs:42]: Actually, [a quick search](https://github.com/search?q=repo%3Awayfair-incubator%2FAeroSharp%20.Any()&type=code) reveals that the only times we're using `Any` are on `IEnumerable<T>`. 

It's incredibly arguable whether or not the collections _should be_ `IEnumerable` and not `IList` (where we could perform such optimization) but that's another problem for another day.

File: src/AeroSharp/DataAccess/Policies/MapConfigurationToMapPolicyMapper.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/Policies/MapConfigurationToMapPolicyMapper.cs:16]: The comments here seem to indicate that they would be mutually exclusive: https://github.com/aerospike/aerospike-client-csharp/blob/80f15b7ffa44af30a8f4e1eeb0a28a1e7cada9a5/AerospikeClient/CDT/MapWriteFlags.cs

I'll see about adding our own validation against that. 😄

File: src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/MapBinParser.cs:42]: > Scratch that. Since we're multi-targeting and supporting .NET 6.0, which only supports up to C# 10, we can't use that form of pattern matching since it's a C# 11 feature. We'd need to do something funky like this, which I don't love
> That said, I'm going to use the count check.

Ah, bummer! Count makes sense, though. :)

File: src/AeroSharp/MapBuilder.cs
Comment (wbaldoumas) [reply src/AeroSharp/MapBuilder.cs:39]: It is intentional, since it visually separates the fields we're setting from input args versus what we're just new-ing up in the constructor. I think it improves readability, but can remove it if that's just me. 😅

File: src/AeroSharp/DataAccess/Policies/MapConfigurationToMapPolicyMapper.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/Policies/MapConfigurationToMapPolicyMapper.cs:16]: Created a validator and doing validation within our builder to catch issues early. LMK what you think!

File: src/AeroSharp/DataAccess/MapConfiguration.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapConfiguration.cs]: Adjusted this to `AllowPartial` for now. Open to other names if you think there's something better we can use.

File: src/AeroSharp/DataAccess/MapAccess/Parsers/MapParserWithSerializer.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Parsers/MapParserWithSerializer.cs]: Updated this to take in bin as an arg and include it in the exception message.

File: src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: I went ahead and added this key type checking, which verifies that the key type must be one of the supported Aerospike scalar data types: https://docs.aerospike.com/server/guide/data-types/scalar-data-types. 

Let me know if this is what you were imagining, or if you think there's a better way of doing this.

Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Initially, I was going to propose something like this:

Since we know all of the types at compile-time, we could take advantage of C# pattern matching. It would be very similar to what the Aerospike client library does [here](https://github.com/aerospike/aerospike-client-csharp/blob/master/AerospikeClient/Value/Value.cs#L235C9-L259).

Something silly like:

```csharp
object integer = 3;
switch(integer) {
    case int myInt:
    case double myDouble:
    default:
        throw new Exception();
}
```

However, I think it really depends on whether or not we think `MapBuilder` might be used in a hot path or not. I think, in all cases, pattern matching would reduce memory allocations but then you lose the convenience of the valid types string you throw in the exception. In a hot path (let's say someone is putting the creation of these into a scoped injection lifetime), it could matter. This does lose the `Type.IsAssignableFrom` check, but that may be okay.

I also think that, based on their documentation, we should include support for `ulong` since they support both unsigned and signed 64 bit integers.

File: src/AeroSharp/DataAccess/Validation/MapConfigurationValidator.cs
Comment (farmerau) [inline src/AeroSharp/DataAccess/Validation/MapConfigurationValidator.cs:12]: Don't we also need to add a rule for `mapConfiguration.UpdateOnly`? I think this only guarantees that `CreateOnly` is false when `UpdateOnly` is true, but not the other way around.

File: src/AeroSharp/MapBuilder.cs
Comment (farmerau) [inline src/AeroSharp/MapBuilder.cs:179]: Missing newline at end of file here

File: tests/AeroSharp.UnitTests/DataAccess/MapAccess/Parsers/MapParserTests.cs
Comment (farmerau) [inline tests/AeroSharp.UnitTests/DataAccess/MapAccess/Parsers/MapParserTests.cs:196]: Missing newline eof

File: tests/AeroSharp.UnitTests/DataAccess/Validation/MapConfigurationValidatorTests.cs
Comment (farmerau) [inline tests/AeroSharp.UnitTests/DataAccess/Validation/MapConfigurationValidatorTests.cs:32]: Is it valid to be neither?

**Review by farmerau — COMMENTED** [review]
Some comments on changes and also got to look at the unit tests

File: src/AeroSharp/DataAccess/Validation/MapConfigurationValidator.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/Validation/MapConfigurationValidator.cs:12]: We don't explicitly need to include that. This validation only runs when `CreateOnly` is `true` and ensures that `UpdateOnly` is set to `false. When `CreateOnly` is set to `false`, it doesn't matter what the value of `UpdateOnly` is. 

The associated unit tests confirm this, but I'd be happy to add the other scenario for the sake of self-documenting code and clarity. What do you think?

File: tests/AeroSharp.UnitTests/DataAccess/Validation/MapConfigurationValidatorTests.cs
Comment (wbaldoumas) [reply tests/AeroSharp.UnitTests/DataAccess/Validation/MapConfigurationValidatorTests.cs:32]: Yeah, in the case when neither are set, the [default policy](https://docs.aerospike.com/apidocs/csharp/html/t_aerospike_client_mapwriteflags) is used, which allows both create and update.

File: src/AeroSharp/DataAccess/Validation/MapConfigurationValidator.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/Validation/MapConfigurationValidator.cs:12]: Ah, this makes sense. I don't think it needs to be explicit anywhere. Sounds good to me!

**Review by farmerau — APPROVED** [review]
LGTM!

**Comment by farmerau** [comment]
Oh, @wbaldoumas not sure if you saw my comment here https://github.com/wayfair-incubator/AeroSharp/pull/170/files#r1277535146

**Comment by wbaldoumas** [comment]
Ah, yeah. I missed that! Let me spend some time thinking about it and respond to you there. Thanks for the feedback. 😄

**Comment by wbaldoumas** [comment]
> Oh, @wbaldoumas not sure if you saw my comment here https://github.com/wayfair-incubator/AeroSharp/pull/170/files#r1277535146

Ah yeah, I missed that. Let me take a look and spend some time thinking about it. I'll respond there!

File: src/AeroSharp/DataAccess/MapConfiguration.cs
Comment (awesor) [inline src/AeroSharp/DataAccess/MapConfiguration.cs:22]: Quick thought here, is it worth either setting the default of ordering to "K-Ordered" instead?  While it's not significant necessarily for this PR (and users should know what they're doing when creating the configs), the main operation of get_by_key is O(log N) in the worst case when key ordered as opposed to O(N) for unordered.

Ref: https://docs.aerospike.com/server/guide/data-types/cdt-map-performance

File: src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Hmm, @farmerau are you thinking we'd remove the type validation in the builder and add this switch statement here in the `MapEntryGenerator.GenerateKey` method, throwing an exception when an unexpected key type is encountered? 

My only hesitation would be that we're deferring the type validation from a spot that _might_ be in the hot path to a spot that is much more likely to be in the hot path, but maybe the pattern matching benefit of reducing memory allocations makes that negligible. 🤔

File: src/AeroSharp/DataAccess/MapConfiguration.cs
Comment (awesor) [reply src/AeroSharp/DataAccess/MapConfiguration.cs:22]: Realizing now that the Aerospike client defaults to unordered as well, so no worries here!

**Review by awesor — APPROVED** [review]

File: src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs
Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Oh, no, I was thinking something like replacing these lines [here](https://github.com/wbaldoumas/AeroSharp/blob/48e8bdf990cd438e6af0f20315decd921a0e6156/src/AeroSharp/MapBuilder.cs#L138-L143):

```csharp
if (!ValidKeyTypes.Any(validKeyType => validKeyType.IsAssignableFrom(keyType)))
{
    var validKeyNames = string.Join(", ", ValidKeyTypes.Select(type => $" {type}"));

    throw new UnsupportedKeyTypeException($"Map keys must be of type{validKeyNames}.");
}
```

Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: We can keep it in this maybe-hot-path but do more to reduce it's impact.

Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Ah, what would that look like, then? We've only got a type parameter `TKey` to work there in the builder, and not a variable to pass into a  `switch` statement that uses pattern matching.

Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: One way I could see improving this is making the valid types a `HashSet<Type>` and doing `.Contains(...)` to make this an ~O(1) operation, rather than using the `.Any(...)`.

```csharp
private static readonly HashSet<Type> ValidKeyTypes = new HashSet<Type>
{
    typeof(long), typeof(double), typeof(string), typeof(byte[]), typeof(bool), typeof(ulong)
};
```

```csharp
private IMapOperator<TKey, TValue> BuildMapOperator<TKey, TValue>()
{
    var keyType = typeof(TKey);

    if (!ValidKeyTypes.Contains(keyType))
    {
        var validKeyNames = string.Join(", ", ValidKeyTypes.Select(type => $"{type}"));

        throw new UnsupportedKeyTypeException($"Map keys must be of type {validKeyNames}.");
    }

    // ...
}
```

Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Oh, you’re totally right. I got lost in my example and didn’t think about the details surrounding it.

I think the HashSet solution is optimal, though!

Comment (awesor) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Is it possible for this to go through and open an issue to add the validation in an upcoming PR?  

I do agree that making sure that our keys are appropriately set is important, but for v.0.0.1 of this Map support, I think users would _hopefully_ read the clear documentation on AS's site about support?

Comment (awesor) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: I've got a PR right on the heels of this that I'd be happy to add to

Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Let me clean this up to use the `HashSet` way of validating the types and then we should be able to merge this in. 

Note that I'm not including a version bump yet or releasing a new version of the package, as we may want to refine this further and add additional map methods before we consider a release, so I'm fine with you adjusting this more in a follow up PR, @awesor.

That said, @farmerau I gave it a go using some `ulong` keys within the integration tests at very high values and I'm seeing them get converted to `-1` on the Aerospike server when read back out. I don't actually think unsigned is supported.

Comment (awesor) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Cool beans - that's really interesting with ulong there!

Comment (farmerau) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: @wbaldoumas Weird on `ulong`-- their docs made me think it was supported. Thanks for testing!

File: src/AeroSharp/DataAccess/MapConfiguration.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapConfiguration.cs:22]: Sounds good. Thanks for double checking!

File: src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs
Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Alright, I pushed up that quick change to swap to a `HashSet` to hold our valid types. Barring any objections, I'll merge this in tomorrow night and hand it on over to you, @awesor, with your follow up PR. Looking forward to our upcoming release! 😄

Comment (wbaldoumas) [reply src/AeroSharp/DataAccess/MapAccess/Generators/MapEntryGeneratorWithSerializer.cs:17]: Also, looks like that last push dismissed both of your reviews, so this will need another stamp of approval from both of you. Thanks for all the feedback!

**Review by farmerau — APPROVED** [review]
LGTM! Thanks for getting this together, @wbaldoumas!

**Review by awesor — APPROVED** [review]

**Pull request merged**
